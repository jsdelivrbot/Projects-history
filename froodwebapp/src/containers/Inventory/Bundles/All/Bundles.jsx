import React from 'react';
import { ConnectedMainContainer } from 'containers';
import { connect } from 'react-redux';
import { BUNDLES_REQUESTS } from 'redux-base/actions';
import { getMainContainerState } from 'utils';
import {
  renderItem,
  getRowClickRedirectLink
} from './bundlesHelpers';

const mapStateToProps = state => getMainContainerState(state, 'bundles');

const mapDispatchToProps = BUNDLES_REQUESTS;

export const Bundles = props => (
  <ConnectedMainContainer
    // autocomplete
    autocompletePlaceholder="Search by sku code"
    renderAutocompleteItem={ renderItem }

    // top buttons
    newButtonVisible
    newButtonText="New Bundle"
    newButtonLink="/inventory/item-bundles/new"

    // table
    actionColumnVisible={ false }
    getRowClickRedirectLink={ getRowClickRedirectLink }

    { ...props }
  />
);

export default connect(mapStateToProps, mapDispatchToProps)(Bundles);
