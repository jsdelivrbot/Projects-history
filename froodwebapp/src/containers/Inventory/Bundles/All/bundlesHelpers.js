import { renderAutocompleteItem } from 'utils';

export const getStatisticsColumns = () => ([
  {
    value: 300,
    description: 'Item Bundles'
  }, {
    value: 20,
    description: 'Since Yesterday',
    isDynamic: true
  }, {
    value: 290,
    description: 'Active'
  }, {
    value: 20,
    description: 'Since Yesterday',
    isDynamic: true
  }, {
    value: 290,
    description: 'Recurring'
  }, {
    value: 15,
    description: 'Since Yesterday',
    isDynamic: true
  }]
);

export const renderItem = renderAutocompleteItem(['id', 'name', 'skuCode'], 'Item Bundle');

export const getRowClickRedirectLink = row => (
  `/inventory/item-bundles/${row.id}`
);
