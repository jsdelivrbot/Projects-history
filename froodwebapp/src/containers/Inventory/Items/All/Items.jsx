import React from 'react';
import { ConnectedMainContainer } from 'containers';
import { connect } from 'react-redux';
import { ITEMS_REQUESTS } from 'redux-base/actions';
import { getMainContainerState } from 'utils';
import {
  renderItem,
  getRowClickRedirectLink
} from './itemsHelpers';

const mapStateToProps = state => getMainContainerState(state, 'items');

const mapDispatchToProps = ITEMS_REQUESTS;

export const Items = props => (
  <ConnectedMainContainer
    // autocomplete
    autocompletePlaceholder="Search by sku code"
    renderAutocompleteItem={ renderItem }

    // top buttons
    newButtonVisible
    newButtonText="New Inventory"
    newButtonLink="/inventory/all-items/new"

    // table
    actionColumnVisible={ false }
    getRowClickRedirectLink={ getRowClickRedirectLink }

    { ...props }
  />
);

export default connect(mapStateToProps, mapDispatchToProps)(Items);
