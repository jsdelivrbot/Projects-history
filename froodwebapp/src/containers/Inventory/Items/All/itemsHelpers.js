import { renderAutocompleteItem } from 'utils';

export const getStatisticsColumns = () => ([
  {
    value: 300,
    description: 'All Items'
  }, {
    value: 20,
    description: 'Since Yesterday',
    isDynamic: true
  }, {
    value: 290,
    description: 'Active'
  }, {
    value: 20,
    description: 'Since Yesterday',
    isDynamic: true
  }, {
    value: 290,
    description: 'Recurring'
  }, {
    value: 15,
    description: 'Since Yesterday',
    isDynamic: true
  }]
);

export const renderItem = renderAutocompleteItem(['id', 'name', 'skuCode'], 'Item');

export const getRowClickRedirectLink = row => (
  `/inventory/all-items/${row.id}`
);
