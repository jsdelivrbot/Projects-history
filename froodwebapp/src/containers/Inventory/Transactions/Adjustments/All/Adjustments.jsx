import React from 'react';
import { ConnectedMainContainer } from 'containers';
import { connect } from 'react-redux';
import { ADJUSTMENTS_REQUESTS } from 'redux-base/actions';
import { getMainContainerState } from 'utils';
import {
  renderItem,
  getStatisticsColumns
} from './adjustmentsHelpers';

const mapStateToProps = state => getMainContainerState(state, 'adjustments');

const mapDispatchToProps = ADJUSTMENTS_REQUESTS;

export const Adjustments = props => (
  <ConnectedMainContainer
    // autocomplete
    autocompletePlaceholder="Search by adjustment name"
    renderAutocompleteItem={ renderItem }

    // top buttons
    newButtonVisible
    newButtonText="New Adjustment"
    newButtonLink="/inventory/adjustments/new"

    // table
    actionColumnVisible={ false }

    // helpers
    getStatisticsColumns={ getStatisticsColumns }
    { ...props }
  />
);

export default connect(mapStateToProps, mapDispatchToProps)(Adjustments);
