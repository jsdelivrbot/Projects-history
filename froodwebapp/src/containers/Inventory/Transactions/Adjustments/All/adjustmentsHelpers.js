import { renderAutocompleteItem } from 'utils';

export const getStatisticsColumns = () => ([
  {
    value: 300,
    description: 'Adjustments'
  }, {
    value: 20,
    description: 'Since Yesterday',
    isDynamic: true
  }, {
    value: 290,
    description: 'Active'
  }, {
    value: 20,
    description: 'Since Yesterday',
    isDynamic: true
  }, {
    value: 290,
    description: 'Recurring'
  }, {
    value: 15,
    description: 'Since Yesterday',
    isDynamic: true
  }]
);

export const renderItem = renderAutocompleteItem(['bin', 'sku', 'warehouse'], 'Adjustment');
