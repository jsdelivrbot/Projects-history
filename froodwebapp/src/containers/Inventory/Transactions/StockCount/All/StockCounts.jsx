import React from 'react';
import { ConnectedMainContainer } from 'containers';
import { connect } from 'react-redux';
import { STOCK_COUNTS_REQUESTS } from 'redux-base/actions';
import { getMainContainerState } from 'utils';
import {
  renderItem,
  getStatisticsColumns,
  getRowClickRedirectLink
} from './stockCountHelpers';

const mapStateToProps = state => getMainContainerState(state, 'stockCounts');

const mapDispatchToProps = STOCK_COUNTS_REQUESTS;

export const StockCounts = props => (
  <ConnectedMainContainer
    // autocomplete
    autocompletePlaceholder="Search by code, delivery date, vendor name"
    renderAutocompleteItem={ renderItem }

    // top buttons
    newButtonVisible
    newButtonText="New Count"
    newButtonLink="/inventory/stock-count/new"

    // table
    actionColumnVisible={ false }
    getRowClickRedirectLink={ getRowClickRedirectLink }

    // helpers
    getStatisticsColumns={ getStatisticsColumns }
    { ...props }
  />
);

export default connect(mapStateToProps, mapDispatchToProps)(StockCounts);
