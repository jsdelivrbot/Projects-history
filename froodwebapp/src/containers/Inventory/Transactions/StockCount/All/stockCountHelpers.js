import { renderAutocompleteItem } from 'utils';

export const getStatisticsColumns = () => ([
  {
    value: 300,
    description: 'Purchases'
  }, {
    value: 20,
    description: 'Since Yesterday',
    isDynamic: true
  }, {
    value: 290,
    description: 'Active'
  }, {
    value: 20,
    description: 'Since Yesterday',
    isDynamic: true
  }, {
    value: 290,
    description: 'Recurring'
  }, {
    value: 15,
    description: 'Since Yesterday',
    isDynamic: true
  }]
);

export const renderItem = renderAutocompleteItem(['code', 'startDate', 'endDate'], 'Stock Count');

export const getRowClickRedirectLink = row => (
  `/inventory/stock-count/${row.id}/${row.code}`
);
