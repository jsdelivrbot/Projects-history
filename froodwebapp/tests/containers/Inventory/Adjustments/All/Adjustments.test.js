import { Adjustments } from 'containers/Inventory/Adjustments/All/Adjustments';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const { describe, expect, it } = global;

describe('Adjustments', () => {
  it('renders the Adjustments component', () => {
    const component = shallow(<Adjustments />);

    expect(component.length).toEqual(1);
  });

  it('capturing Snapshot of Adjustments', () => {
    const component = shallow(<Adjustments />);
    const view = toJson(component);
    const actual = expect(view);

    actual.toMatchSnapshot();
  });
});
