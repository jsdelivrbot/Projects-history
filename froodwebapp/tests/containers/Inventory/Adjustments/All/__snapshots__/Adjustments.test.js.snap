// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adjustments capturing Snapshot of Adjustments 1`] = `
<Connect(MainContainer)
  actionColumnVisible={false}
  autocompletePlaceholder="Search by adjustment name"
  getStatisticsColumns={[Function]}
  newButtonLink="/inventory/adjustments/new"
  newButtonText="New Adjustment"
  newButtonVisible={true}
  renderAutocompleteItem={[Function]}
/>
`;
